sdk/search/Azure.Search/src/SearchIndexClient.cs

```diff
@@ -207,61 +215,1398 @@ public class SearchIndexClient
             Pipeline = pipeline;
             Version = version;
 
-            Operations = new DocumentsClient(
+            Protocol = new DocumentsRestClient(
                 ClientDiagnostics,
                 Pipeline,
                 endpoint.ToString(),
                 IndexName,
                 Version.ToVersionString());
         }
+        #endregion ctors
 
+        #region GetDocumentCount
```

Comment:
IMO: 🤮

Reply:
The `region` or the name or ...?  I definitely do the `region` thing because a lot of the Storage files were MASSIVE.